### ISSUE TITLE
Custom dropdown Data Source

### ISSUE THREAD

**Issue by mfarazkhan** [andrewtelnov/surveyjs#51]
Hi,

i am trying to achieve a custom data source for dropdown components, which will help dropdown to be populated from a different data source rather than defining in the editor.
please have a look at the below changes, confirm if this is the right place or will it cause some other problems.
 Object.defineProperty(n.prototype, "choices", {
            get: function () {
                debugger;
                //"dataServiceURL" is a new property in dropdown property bag.
                if (this.dataServiceURL != undefined) {
                    this.choicesValues = []; //reinitialize the Array
                    //loading with a different source.
                    this.choices = ["1|1st modified Item", "2|2nd Modified Item"];
                    console.log(this.choicesValues);
                    return this.choicesValues;
                } else {
                    return this.choicesValues
                }
            },
            set: function (t) {
                debugger;
                e.ItemValue.setData(this.choicesValues, t)
            }

**Comment by andrewtelnov** [comment]
Please refer to this issue: https://github.com/andrewtelnov/surveyjs/issues/67

**Comment by andrewtelnov** [comment]
Hi, 
I have implemented the first variant of the feature. Could you check it out?
I am going to work on property editor now. I also thinking about having a parameter support in the url. For example, when you have the employee list and you may show it only after the person select the department. So you do not show the full list, but filtered by a parameter, previous answered question value. Do you need this?

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
Please, take a look at: http://surveyjs.org/examples/questiontype-dropdownrestfull.html

**Comment by PankajKumawat125** [comment]
Can i call a function in js instead of URL ?

**Comment by andrewtelnov** [comment]
@PankajKumawat125 you may get a question and set the choices:
var question = survey.getQuestionByName("myQuestionName");
```
question.choices = [1, 2];
//or 
question.choices = [{value: 1, text: "item 1"}, {value: 2, text: "item 2"}];
```

Thank you,
Andrew

**Comment by PankajKumawat125** [comment]
@andrewtelnov  Perfect. work fine.
Can you help me with one more thing ?
If i want to pre-select one of the value and show the dropdown then How do i do that?

**Comment by andrewtelnov** [comment]
@PankajKumawat125 survey.setValue("YourQuestionName", defaultValue); [example]( https://surveyjs.io/Examples/Library/?id=survey-data)
or use the defaultValue property in the question, defaultValue: yourDefaultValue.

Please go through our demos. You are asking very basic questions.

Thank you,
Andrew

**Comment by PankajKumawat125** [comment]
@andrewtelnov  Well, I have already done that. i was just going to delete my 2nd question.
and
In this case it was fetching dynamically, that is why i asked you this question!
Anyways thank you so much for your help
